packages/core/src/lib/implementation/persist.spec.ts

```diff
@@ -34,70 +34,86 @@ const readReport = (format: 'json' | 'md') => {
   }
 };
 
-const config = dummyConfig(MEMFS_VOLUME);
+const dummyReport = report();
+const dummyConfig = config(outputDir);
 let logs: string[] = [];
 
+const resetFiles = async () => {
+  vol.reset();
+  vol.fromJSON(
+    {
+      [reportPath('json')]: '',
+      [reportPath('md')]: '',
+    },
+    MEMFS_VOLUME,
+  );
+  unlinkSync(reportPath('json'));
+  unlinkSync(reportPath('md'));
+};
+const setupConsole = async () => {
+  logs = [];
+  mockConsole(msg => logs.push(msg));
+};
+const teardownConsole = async () => {
+  logs = [];
+  unmockConsole();
+};
+
+// @TODO refactor away from snapshots in favour of disc space and readability
```

Comment:
I think it's valuable to have the snapshots, but I would only have one per each format. That meangs that `should persist all formats` and `should persist some formats` wouldn't have `.toMatchSnapshot` as well, because the content is already tested elsewhere. That would also help with the file not being too big.

Reply:
Actually, I see the actual formatting is already tested in `utils`. In that case, maybe it's enough to have it there? We should have the most detailed test "at source" (less setup needed, easier to test many different scenarios), wrapper functions don't have to test in so much detail.

Reply:
Done. Good point!